MonalikaPatnaik/UMatter
# Initial Setup - ESLint, Prettier, CommitLint, and Husky ⚡ for React Project

## Fixes Issue

#1075 


## Changes proposed


**GitHub Pull Request Description: Initial Setup - ESLint, Prettier, CommitLint, and Husky ⚡ for React Project**

**Description:**
This pull request introduces essential development tools to improve code quality and maintain consistency in our React project. The following changes have been made:

1. **ESLint Integration:**
   ESLint is a powerful linting tool that helps us identify and fix potential errors, enforce coding standards, and maintain a consistent code style across our project. The ESLint configuration has been set up based on best practices for React development.

2. **Prettier Integration:**
   Prettier is an opinionated code formatter that automatically formats our code according to a predefined set of rules. By integrating Prettier, we ensure that the entire codebase adheres to a unified and clean style.

3. **CommitLint Implementation:**
   CommitLint enforces a standard format for our commit messages. This will help us maintain a well-organized commit history and improve collaboration among team members.

4. **Husky Setup:**
   Husky is a powerful Git hook manager that allows us to define pre-commit and pre-push hooks. We have configured Husky to run ESLint and Prettier before each commit, preventing the commit of code that does not meet our defined standards.

**Why these changes are important:**
- **Consistent Code Style:** Enforcing coding standards with ESLint and Prettier will lead to a more consistent codebase, making it easier for team members to understand and work on each other's code.
- **Early Error Detection:** ESLint will catch potential errors and issues early in the development process, reducing the likelihood of bugs and improving code reliability.
- **Better Collaboration:** CommitLint ensures that commit messages are clear and informative, enhancing collaboration and making it easier to track changes.
- **Automatic Formatting:** With Prettier, we no longer need to worry about manual code formatting, saving time and effort while maintaining a clean and uniform codebase.
- **Preventing Bad Commits:** Husky runs ESLint and Prettier before each commit, preventing the introduction of poorly formatted or erroneous code into the repository.

**How to Test:**
1. Clone this branch to your local development environment.
2. Run `npm install` to install the necessary dependencies.
3. Make changes to the code, deliberately introducing some linting errors or code style violations.
4. Attempt to commit the changes using `git commit` command.
5. Observe that Husky prevents the commit due to ESLint and Prettier errors, ensuring only properly formatted and linted code can be committed.

Let's merge this pull request to kickstart our development with improved code quality and better collaboration practices. If you have any suggestions or feedback, feel free to share. Happy coding! ⚡🚀



## Check List (Check all the applicable boxes) 

- [ ] My change requires changes to the documentation.
